addons/mrp/tests/test_byproduct.py

```diff
@@ -407,3 +407,49 @@ def test_check_byproducts_cost_share_02(self):
         mo.move_byproduct_ids[0].quantity_done = 1
         mo.button_mark_done()
         self.assertEqual(mo.state, 'done')
+
+    def test_3_steps_byproduct(self):
+        """ Test that non-bom byproducts are correctly pushed from
+        post-production to the stock location in 3-steps manufacture. """
+        self.warehouse.manufacture_steps = 'pbm_sam'
+        self.env.user.groups_id += self.env.ref('mrp.group_mrp_byproducts')
+        component = self.env['product.product'].create({
+            'name': 'Old Blood',
+            'type': 'product'
+        })
+        self.env['stock.quant']._update_available_quantity(component, self.warehouse.lot_stock_id, 1)
+        final_product = self.env['product.product'].create({
+            'name': 'Insight',
+            'type': 'product'
+        })
+        byproduct = self.env['product.product'].create({
+            'name': 'Eyes on the Inside',
+            'type': 'product'
+        })
```

Comment:
```suggestion
        component, final_product, byproduct = self.env['product.product'].create([{
            'name': name,
            'type': 'product'
        } for name in ['Old Blood', 'Insight', 'Eyes on the Inside']])
        self.env['stock.quant']._update_available_quantity(component, self.warehouse.lot_stock_id, 1)
```
When possible, batch record creation :slightly_smiling_face: